src/components/menu/menu.tsx

```diff
@@ -61,61 +58,39 @@ export class Menu {
      * Is emitted when a menu item is selected.
      */
     @Event()
-    private select: EventEmitter<ListItem>;
-
-    @Element()
-    private element: HTMLElement;
+    private select: EventEmitter<ListItem | ListItem[]>;
 
-    private menu: MDCMenu;
-    private listRenderer = new ListRenderer();
+    private portalId: string;
 
     constructor() {
-        this.handleMdcSelected = this.handleMdcSelected.bind(this);
-        this.handleMdcClosed = this.handleMdcClosed.bind(this);
-    }
-
-    public componentDidLoad() {
-        const menuElement = this.element.shadowRoot.querySelector('.mdc-menu');
-        this.menu = new MDCMenu(menuElement);
-        this.menu.listen('MDCMenu:selected', this.handleMdcSelected);
-        this.menu.listen('MDCMenuSurface:closed', this.handleMdcClosed);
-
-        this.menu.setAnchorCorner(Corner.TOP_START);
-        this.menu.quickOpen = true;
-    }
-
-    public componentDidUnload() {
-        this.menu.unlisten('MDCMenu:selected', this.handleMdcSelected);
-        this.menu.unlisten('MDCMenuSurface:closed', this.handleMdcClosed);
-        this.menu.destroy();
+        this.portalId = createRandomString();
     }
 
     public render() {
-        const config: ListRendererConfig = {
-            isMenu: true,
-            iconSize: 'small',
-            isOpen: this.open,
-            badgeIcons: this.badgeIcons,
-            type: 'selectable',
-        };
         return (
-            <div class="mdc-menu-surface--anchor">
+            <div class="mdc-menu-surface--anchor" onClick={this.onTriggerClick}>
                 <slot name="trigger">{this.renderTrigger()}</slot>
-                <div class="mdc-menu mdc-menu-surface" tabindex="-1">
-                    {this.listRenderer.render(this.items, config)}
-                </div>
-                {this.disabled ? <div class="menu-disabled" /> : null}
+                <limel-portal
+                    visible={this.open}
+                    containerId={this.portalId}
+                    openDirection={this.openDirection}
+                >
+                    <limel-menu-surface
+                        open={this.open}
+                        onDismiss={this.onClose}
+                    >
```

Comment:
This version doesn't seem to be using a scrim, so how do we detect clicks outside the menu to make the menu close?